radix-engine-tests/tests/address.rs

```diff
@@ -0,0 +1,259 @@
+use radix_engine::errors::{RuntimeError, SystemError};
+use radix_engine::types::*;
+use scrypto_unit::*;
+use transaction::builder::ManifestBuilder;
+
+#[test]
+fn get_global_address_in_local_should_fail() {
+    // Arrange
+    let mut test_runner = TestRunner::builder().build();
+    let package_address = test_runner.compile_and_publish("./tests/blueprints/address");
+    let manifest = ManifestBuilder::new()
+        .lock_fee(FAUCET_COMPONENT, 10.into())
+        .call_function(
+            package_address,
+            "CalledComponent",
+            "create",
+            manifest_args!(),
+        )
+        .build();
+    let receipt = test_runner.execute_manifest(manifest, vec![]);
+    let called_component = receipt.expect_commit(true).new_component_addresses()[0];
+
+    // Act
+    let manifest = ManifestBuilder::new()
+        .lock_fee(FAUCET_COMPONENT, 10.into())
+        .call_function(
+            package_address,
+            "MyComponent",
+            "get_address_in_local",
+            manifest_args!(called_component),
+        )
+        .build();
+    let receipt = test_runner.execute_manifest(manifest, vec![]);
+
+    // Assert
+    receipt.expect_specific_failure(|e| {
+        matches!(
+            e,
+            RuntimeError::SystemError(SystemError::GlobalAddressDoesNotExist)
+        )
+    });
+}
+
```

Comment:
Am I right in understanding your rules?

1. `Function/NonSelfMethod` -> `LocalComponent` -> `Runtime::get_global_address`   ===> Failure
2. `GlobalComponent::Method` -> `LocalComponent` -> `Runtime::get_global_address` ===> Success
3. `GlobalComponent::Method` -> `OwnedComponent` -> `Runtime::get_global_address` ===> Success

I'm not feeling comfortable reading 1 & 2 at the same time.

Reply:
Ah good point! 2 SHOULD fail but I think current implementation will let it pass. Let me verify.